### PR TITLE
Replaced floatval() with type cast (float) as part of issue #17432.

### PR DESCRIPTION
**Pull request by annafoldvari**
See #17432

#### Changes proposed in this Pull Request:

* Replaced floatval() with type cast (float) as part of issue #17432.

#### Testing instructions:

* Go to '..'
*

#### Proposed changelog entry for your changes:

* Replaced floatval() with type cast (float) as part of issue #17432.

### CODE REVIEW

**Review by kraftbj — DISMISSED** [review]
Looks good.

**Review by kbrown9 — APPROVED** [review]
The changes look good to me! I manually tested a few of them (heartbeat, simple payments, presentations, and sync) and everything worked as expected. Thanks for fixing these!

File: class.json-api-endpoints.php [hunk 1]
```diff
@@ -834,7 +834,7 @@ function document( $show_description = true ) {
 <?php endif; ?>
 
 		<?php if ( true === $this->deprecated ) { ?>
-<p><strong>This endpoint is deprecated in favor of version <?php echo floatval( $this->new_version ); ?></strong></p>
+<p><strong>This endpoint is deprecated in favor of version <?php echo (float) $this->new_version ; ?></strong></p>
```
Comment (jeherve) [inline class.json-api-endpoints.php:837]:
```suggestion
<p><strong>This endpoint is deprecated in favor of version <?php echo (float) $this->new_version; ?></strong></p>
```

File: modules/custom-css/custom-css/preprocessors/lessc.inc.php [hunk 2]
```diff
@@ -1396,7 +1396,7 @@ protected function funcToColor($func) {
 			$i = 0;
 			foreach ($rawComponents as $c) {
 				$val = $this->reduce($c);
-				$val = isset($val[1]) ? floatval($val[1]) : 0;
+				$val = isset($val[1]) ? (float) $val[1] : 0;
```
Comment (jeherve) [inline modules/custom-css/custom-css/preprocessors/lessc.inc.php:1399]:
```suggestion
				$val = isset( $val[1] ) ? (float) $val[1] : 0;
```

**Review by jeherve — CHANGES_REQUESTED** [review]
This looks good. I only have minor comments about linting errors that will trip our pre-commit hooks.

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
Internal reference: r215258-wpcom

**Pull request merged**